### PR TITLE
Update var to let/const in docs/Web/JavaScript/Reference/Global_Objects/string/charcodeat

### PR DESCRIPTION
**Pull request by samantharamon**
#### Summary
Update var to let/const.

#### Motivation
- Write the Docs 2022 Writing Day project.
- Conform to current specifications.

#### Supporting details
None.

#### Related issues
N/A

#### Metadata

This PR…

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [X] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md [hunk 1]
```diff
@@ -127,12 +127,11 @@ function fixedCharCodeAt(str, idx) {
...
   str += '';
-  var code,
-      end = str.length;
+  const end = str.length;
 
-  var surrogatePairs = /[\uD800-\uDBFF][\uDC00-\uDFFF]/g;
+  const surrogatePairs = /[\uD800-\uDBFF][\uDC00-\uDFFF]/g;
   while ((surrogatePairs.exec(str)) != null) {
-    var li = surrogatePairs.lastIndex;
+    let li = surrogatePairs.lastIndex;
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md:134]:
```suggestion
    const li = surrogatePairs.lastIndex;
```

File: files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md [hunk 2]
```diff
@@ -145,12 +144,11 @@ function knownCharCodeAt(str, idx) {
...
 
-  code = str.charCodeAt(idx);
+  const code = str.charCodeAt(idx);
 
-  var hi, low;
   if (0xD800 <= code && code <= 0xDBFF) {
-    hi = code;
-    low = str.charCodeAt(idx + 1);
+    let hi = code;
+    let low = str.charCodeAt(idx + 1);
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md:151]:
```suggestion
    const hi = code;
    const low = str.charCodeAt(idx + 1);
```

**Review by wbamberg — DISMISSED** [review]
Thank you for the PR, @samantharamon !

I think a couple of those `let`s could be `const`s.

File: files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md (hunk 2 above)
Comment (samantharamon) [reply files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md:151]: Thanks for the suggestion, @wbamberg!

File: files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md (hunk 1 above)
Comment (samantharamon) [reply files/en-us/web/javascript/reference/global_objects/string/charcodeat/index.md:134]: Thanks for the suggestion, @wbamberg!

**Review by queengooborg — APPROVED** [review]
Thank you, this is LGTM!

**Pull request merged**